packages/mobile/src/identity/contactMapping.ts

```diff
@@ -273,10 +283,18 @@ const isValidNon0Address = (address: string) =>
 export function getAddressFromPhoneNumber(
   e164Number: string,
   e164NumberToAddress: E164NumberToAddressType,
-  secureSendPhoneNumberMapping: SecureSendPhoneNumberMapping
+  secureSendPhoneNumberMapping: SecureSendPhoneNumberMapping,
+  type: TokenTransactionType,
+  requesterAddress?: string
 ): string | null | undefined {
   const addresses = e164NumberToAddress[e164Number]
 
+  // If there are no verified addresses for the number and it is a payment request,
+  // use the requester's given address
+  if (!addresses && type === TokenTransactionType.PayRequest && requesterAddress) {
```

Comment:
Can we remove the `type` param and just return requesterAddress if `!address && requesterAddress`?
My concern is the unnecessary coupling this change introduces. Currently the contact mapping code is unaware of the concept of token types and it's desirable to keep these sub-systems self-contained.